temporal-normalizer/src/main/java/edu/illinois/cs/cogcomp/temporal/normalizer/main/TemporalChunkerAnnotator.java

```diff
@@ -0,0 +1,319 @@
+     */
+    public void addDocumentCreationTime(String date) {
+        SimpleDateFormat f = new SimpleDateFormat("dd-MMM-yyyy");
+        try {
+            this.dct = f.parse(date);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Normalize temporal phrase
+     * @param temporal_phrase
+     * @return
+     * @throws Exception
+     */
+    private String heidelTimeNormalize(Constituent temporal_phrase) throws Exception {
+        // If user didn't specify document creation date, use the current date
+        if (this.dct == null) {
+            this.dct = new Date();
+        }
+
+        String xml_res = this.heidelTime.process(temporal_phrase.toString(), this.dct);
+
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document document = builder.parse(new InputSource(new StringReader(xml_res)));
+
+        Element rootElement = document.getDocumentElement();
+        String res = recurseNormalizedTimeML(rootElement);
+        return res;
+    }
+
+
+    private String recurseNormalizedTimeML(Node node) {
+        // Base case: return empty string
+        if (node == null) {
+            return "";
+        }
+
+        // Iterate over every node, if the node is a TIMEX3 node, then concatenate all its attributes, and recurse
+        NodeList nodeList = node.getChildNodes();
+        for (int i = 0; i < nodeList.getLength(); i++) {
+            Node currentNode = nodeList.item(i);
+            if (currentNode.getNodeType() == Node.ELEMENT_NODE && currentNode.getNodeName().indexOf("TIMEX3")!=-1) {
+                //calls this method for all the children which is Element
+                NamedNodeMap attrs = currentNode.getAttributes();
+                String attrPair = "";
+                for (int j = 0; j < attrs.getLength(); j++) {
+                    attrPair += "["
+                            + attrs.item(j).getNodeName() + "="
+                            + attrs.item(j).getNodeValue() + "]" ;
+                }
+                return attrPair + recurseNormalizedTimeML(currentNode);
+            }
+        }
+        return "";
+    }
+
+    @Override
+    public String getViewName() {
+        return ViewNames.TIMEX3;
+    }
+
+    /**
+     * Can be used internally by {@link edu.illinois.cs.cogcomp.annotation.AnnotatorService} to
+     * check for pre-requisites before calling any single (external)
+     * {@link edu.illinois.cs.cogcomp.annotation.Annotator}.
+     *
+     * @return The list of {@link edu.illinois.cs.cogcomp.core.datastructures.ViewNames} required by
+     *         this ViewGenerator
+     */
+    @Override
+    public String[] getRequiredViews() {
+        return new String[] {ViewNames.POS};
+    }
+
+
+    public static void main(String []args) throws Exception{
+        ResourceManager rm = new ResourceManager("./conf/chunker_config.props");
```

Comment:
The config should be programmatic (i.e. extend Configurator) and so a default constructor can be used.  Exceptions: when the user wants to override a default option, or where the class requires a proprietary resource like an LDC corpus, and you want to force the user to specify it.